### PR TITLE
Update MAC documentation and improve code tidiness

### PR DESCRIPTION
**Pull request by davidvincze**
Fixing #128

### CODE REVIEW

File: src/t_cose_mac_compute.c [hunk 1]
```diff
@@ -23,57 +23,11 @@
...
-    return return_value;
-}
-
 /*
- * Public function. See t_cose_mac.h
+ * Public function. See t_cose_mac_compute.h
  */
 enum t_cose_err_t
-t_cose_mac_encode_parameters(struct t_cose_mac_calculate_ctx *me,
+t_cose_mac_encode_parameters(struct t_cose_mac_calculate_ctx *context,
```
Comment (laurencelundblade) [inline src/t_cose_mac_compute.c:30]: Prefer "me" in the implementation.

**Review by laurencelundblade — CHANGES_REQUESTED** [review]
Just a couple of tweaks requested. Thank you for doing this, David!

LL

File: inc/t_cose/t_cose_mac_compute.h [hunk 2]
```diff
@@ -219,63 +287,67 @@ t_cose_mac_encode_tag(struct t_cose_mac_calculate_ctx *context,
 
 
 static inline void
-t_cose_mac_compute_init(struct t_cose_mac_calculate_ctx *me,
+t_cose_mac_compute_init(struct t_cose_mac_calculate_ctx *context,
```
[the remark this answers is not in the document]
Comment (davidvincze) [reply inc/t_cose/t_cose_mac_compute.h]: I've updated it in mac_compute.h/c and mac_validate.h/.c.

File: inc/t_cose/t_cose_mac_compute.h [hunk 3]
```diff
@@ -211,6 +167,118 @@ t_cose_mac_encode_tag(struct t_cose_mac_calculate_ctx *context,
...
+ */
+static enum t_cose_err_t
+t_cose_mac_compute_detached(struct t_cose_mac_calculate_ctx *context,
+                            struct q_useful_buf_c            aad,
+                            struct q_useful_buf_c            datached_payload,
+                            struct q_useful_buf              out_buf,
+                            struct q_useful_buf_c           *result);
+
+
+/**
```
[the remark this answers is not in the document]
Comment (davidvincze) [reply inc/t_cose/t_cose_mac_compute.h:259]: Updated.

**Comment by davidvincze** [comment]
Thank you, Laurence, for the feedback, I've updated the patch.

**Pull request merged**